src/vector/mod.rs

```diff
@@ -127,6 +135,66 @@ pub trait VectorOps {
         derivation_path: DerivationPath,
         metadata: &AlloyMetadata,
     ) -> Result<Vec<u8>, AlloyError>;
+
+    /// Rotates vectors from the in-rotation secret for their secret path to the current secret.
+    /// This can also be used to rotate data from one tenant ID to a new one, which most useful when a tenant is
+    /// internally migrated.
+    ///
+    /// WARNINGS:
+    ///     * this involves decrypting then encrypting vectors. Since the vectors are full of floating point numbers,
+    ///       this process is lossy, which will cause some drift over time. If you need perfectly preserved accuracy
+    ///       store the source vector encrypted with `standard` next to the encrypted vector. `standard` decrypt
+    ///       that, `vector` encrypt it again, and replace the encrypted vector with the result.
+    ///     * only one metadata and new tenant ID argument means each call to this needs to have one tenant's vectors.
+    async fn rotate_vectors(
+        &self,
+        encrypted_vectors: EncryptedVectors,
```

Comment:
The SaaS Shield version of this function will make 2 TSP calls for every vector passed in instead of using the `derive_keys_many_paths` to reduce them. We don't yet have batch_encrypt or batch_decrypt, but I feel like this should be more of a batchier function than it currently is.

Reply:
In order to support this, we could have a private trait that implemented more batchy internal functions that we need, but don't want to export to the caller. Or we could implement this functionality in each trait impl instead of having a blanket impl. There wouldn't be much duplication I don't think, but I haven't sketched it all the way out.

Reply:
We want batch calls to be public anyway so I think we'd consume those vs adding internal functions.